tests/unit/evaluation/SplittingStrategy_unittest.cc

```diff
@@ -296,3 +296,90 @@ TEST(SplittingStrategy,LOO)
 		SG_UNREF(splitting);
 	}
 }
+
+TEST(SplittingStrategy, timeseries_subset_linear_splits)
+{
+	index_t num_labels, num_subsets, min_subset_size, base_size;
+	index_t runs = 10;
+
+	while (runs-- > 0)
+	{
+		num_labels = CMath::random(50, 150);
+		num_subsets = CMath::random(1, 5);
+		min_subset_size = CMath::random(1, 6);
+		base_size = num_labels / num_subsets;
+
+		CRegressionLabels* labels = new CRegressionLabels(num_labels);
+		for (index_t i = 0; i < num_labels; ++i)
+			labels->set_label(i, CMath::random(-10.0, 10.0));
+
+		CTimeSeriesSplitting* splitting =
+		    new CTimeSeriesSplitting(labels, num_subsets);
+
+		splitting->set_min_subset_size(min_subset_size);
+		splitting->build_subsets();
+
+		for (index_t i = 0; i < num_subsets; ++i)
+		{
+			SGVector<index_t> subset = splitting->generate_subset_indices(i);
+			SGVector<index_t> inverse = splitting->generate_subset_inverse(i);
+
+			/* Subset size should be atleat min_subset_size */
+			EXPECT_GE(subset.vlen, splitting->get_min_subset_size());
+
+			/* check the splitting is linear */
+			EXPECT_TRUE(
+			    inverse.vlen % base_size == 0 ||
+			    inverse.vlen == num_labels - min_subset_size);
+		}
+
+		SG_UNREF(splitting);
+	}
+}
+
+TEST(SplittingStrategy, timeseries_subsets_future_leak)
+{
+	index_t num_labels, num_subsets, min_subset_size;
+	index_t runs = 10;
+
+	while (runs-- > 0)
+	{
+		num_labels = CMath::random(50, 150);
+		num_subsets = CMath::random(1, 5);
+		min_subset_size = CMath::random(1, 7);
+
+		CRegressionLabels* labels = new CRegressionLabels(num_labels);
```

Comment:
again, all this should be moved to some helper function. the test is bloated and hard to read right now